src/tensors/tensor.jl

```diff
@@ -36,9 +36,32 @@ struct TensorMap{S<:IndexSpace, N₁, N₂, I<:Sector, A<:Union{<:DenseMatrix,Se
 end
 #! format: on
 
+"""
+    Tensor{S<:IndexSpace, N, I<:Sector, A, F₁, F₂} = TensorMap{S, N, 0, I, A, F₁, F₂}
+
+Specific subtype of [`AbstractTensor`](@ref) for representing tensors whose data is stored
+in blocks of some subtype of `DenseMatrix`.
+
+A `Tensor{S, N, I, A, F₁, F₂}` is actually a special case `TensorMap{S, N, 0, I, A, F₁, F₂}`,
+i.e. a tensor map with only a non-trivial output space.
+"""
 const Tensor{S<:IndexSpace,N,I<:Sector,A,F₁,F₂} = TensorMap{S,N,0,I,A,F₁,F₂}
+"""
+    TrivialTensorMap{S<:IndexSpace, N₁, N₂, A<:DenseMatrix} = TensorMap{S, N₁, N₂, Trivial, 
+                                                                        A, Nothing, Nothing}
+
+A special case of [`TensorMap`](@ref) for representing tensor maps with trivial symmetry,
+i.e., whose `sectortype` is `Trivial`.
+"""
 const TrivialTensorMap{S<:IndexSpace,N₁,N₂,A<:DenseMatrix} = TensorMap{S,N₁,N₂,Trivial,A,
```

Comment:
It seems like my code is not very consistent with which subtype relations are imposed in these constants/aliases. In this one, `A<:DenseMatrix`, but in the `Tensor` definition it is not.

Reply:
I think it might be the most convenient to not impose anything for the aliases. The restrictions will still apply, as the original type has them, and if ever they need to change, this needs to only happen at a single point in the code.

Reply:
Makes sense. I dropped the restrictions that are just duplicates of the original `TensorMap` ones, and it seems all I'm left with is the `A<:DenseMatrix` in `TrivialTensorMap`. As far as I understand the storagetype for a `Tensor` can still be a `SectorDict{<:Sector,<:DenseMatrix}` (with a single entry?), so I don't think this one should have an additional subtype restriction?

Reply:
Thanks. This looks good now. As for the `A` parameter, we are considering to revise the `TensorMap` structure quite drastically, so this will probably change anyway in the (hopefully near) future.